### DISCUSSION TITLE
having the local dev server run custom resolvers that are hosted in separate lambda's

### DISCUSSION THREAD

**Discussion by tomschut** [maoosi/prisma-appsync#126] [Ideas]
Hi,

I'm not sure if this is already part of the implementation, but I'm unable to find it. 
We'd like to run some custom resolvers which will eventually run in their own lambda's. I think it's quite doable to change the cdk deploy script to include these lambda's as datasources for appsync but I'm not sure if there's any way of including them in the local dev server. Is this possible?

Kind regards,
Tom

**Comment by maoosi** [comment]
Creating and running Custom Resolvers on the local dev server is possible and documented here: 
https://github.com/maoosi/prisma-appsync/blob/dev/docs/features/resolvers.md

However, running multiple lambda functions is not supported as it is quite a specific use case. The current implementation of the local dev server takes a single `handler.ts` file as an input and will pass all requests to that same handler. [See full implementation here](https://github.com/maoosi/prisma-appsync/blob/dev/packages/server/src/index.ts).

It might be possible to adapt the current options to support multiple handlers with some sort of filters to redirect queries to the right handler based on query names - but it would be quite manual to setup (the local dev server is completely independent from your CDK files).

**Reply by tomschut** [reply]
Thanks for your explanation. I suppose that what I'm asking can be easily worked around by including the function calls that are on AWS housed in their own lambda's as custom resolvers only for the dev server, so that they follow the normal pattern and we can test with them, which might be more feasible.

**Comment by maoosi** [comment]
Should now be possible with https://github.com/maoosi/prisma-appsync/pull/147 - Will publish some docs along with the next release.
